Automattic/pocket-casts-android
# Fixing toolbar memory leaks

## Description

These changes are from the leaks found when running LeakCanary. 

## Testing Instructions
1. Tap the "Profile" tab
2. Tap the "Account" button
3. ✅ Verify there is a toolbar with a back button
4. Tap the "Change password" section
5. ✅ Verify there is a toolbar with a back button
6. Tap the back button to return back to the Account page
7. Tap the back button to return back to the Profile page
8. Tap on the "Podcasts" tab
9. Tap on a podcast
10. Tap on the settings icon
11. Tap on Playback Effects
12. ✅ Verify there is a toolbar with a back button
13. Tap on the back button
